lib/overcommit/hook_context/pre_push.rb

```diff
@@ -18,17 +18,16 @@ def pushed_refs
     end
 
     def modified_files
-      @modified_files ||= Overcommit::GitRepo.modified_files(refs: ref_range)
+      @modified_files ||= pushed_refs.each_with_object(Set.new) do |pushed_ref, set|
+        set.merge(pushed_ref.modified_files)
+      end.to_a
```

Comment:
```
require 'benchmark'
require 'set'

iterations = 10
data = 100.times.map do
  Array.new(1_000) { rand(0..1_000) }
end

Benchmark.bmbm do |x|
  x.report('array.uniq') do
    iterations.times do
      data.uniq
    end
  end

  x.report('array.each_with_object{ merge }.to_a') do
    iterations.times do
      data.each_with_object(Set.new) do |datum, set|
        set.merge(datum)
      end.to_a
    end
  end
end

puts 'Are they equivalent?'
puts data.uniq == data.each_with_object(Set.new) { |datum, set| set.merge(datum) }.to_a

Rehearsal ------------------------------------------------------------------------
array.uniq                             0.250000   0.000000   0.250000 (  0.251087)
array.each_with_object{ merge }.to_a   0.140000   0.000000   0.140000 (  0.138263)
--------------------------------------------------------------- total: 0.390000sec

                                           user     system      total        real
array.uniq                             0.240000   0.000000   0.240000 (  0.244006)
array.each_with_object{ merge }.to_a   0.130000   0.000000   0.130000 (  0.138962)
Are they equivalent?
false
```

Reply:
Seems like using `Set` with `each_with_object` is faster than `Array` with `uniq`.